chaoscenter/authentication/api/handlers/rest/capabilities_handler.go

```diff
@@ -0,0 +1,25 @@
+package rest
+
+import (
+	"github.com/gin-gonic/gin"
+	response "github.com/litmuschaos/litmus/chaoscenter/authentication/api/handlers"
+	"github.com/litmuschaos/litmus/chaoscenter/authentication/pkg/utils"
+)
+
+// GetCapabilities 		godoc
+//
+//	@Summary		Get capabilities of Auth Server.
+//	@Description	Returns capabilities that can be leveraged by frontend services to toggle certain features.
+//	@Tags			CapabilitiesRouter
+//	@Accept			json
+//	@Produce		json
+//	@Failure		500	{object}	response.ErrServerError
+//	@Success		200	{object}	response.CapabilitiesResponse{}
+//	@Router			/capabilities [get]
+func GetCapabilities() gin.HandlerFunc {
```

Comment:
`GetCapabilities()` is associated with Dex, so how about using exist [dex_auth_handler](https://github.com/litmuschaos/litmus/blob/master/chaoscenter/authentication/api/handlers/rest/dex_auth_handler.go) not creating a new `routes` and `handlers`?

in your suggestion, it would be great if ...
`GetCapabilities()` -> `DexEnabled()`
`router.GET("/capabilities", rest.GetCapabilities())` -> `router.GET("/dex/enabled", rest.DexEnabled())`

If you adopt my suggestion, you also need to change files

from
https://github.com/litmuschaos/litmus/blob/d176b92e26b7cbedfe805439788fc9631f6536be/chaoscenter/authentication/api/main.go#L167-L169

to 
```go
routes.DexRouter(app, applicationService)
```

and in [this file](https://github.com/litmuschaos/litmus/blob/master/chaoscenter/authentication/api/routes/dex_router.go), you need to change like this

```go
func DexRouter(router *gin.Engine, service services.ApplicationService) {
        router.GET("/dev/enabled", rest.DexEnabled())
        if utils.DexEnabled {
                router.GET("/dex/login", rest.DexLogin())
	        router.GET("/dex/callback", rest.DexCallback(service))
        }
}
```

how do you think? let me know if you have any comments 😄

Reply:
Thank you, @namkyu1999, for your detailed comment. My aim was to make the capabilities endpoint generic so it can tell the frontend about different features the auth server has. This way, if we need to add more features later or allow additional customisation, we can use the same endpoint to expose more information. I took an inspiration from [ArgoCD](https://github.com/argoproj/argo-cd/blob/master/server/settings/settings.go#L39). 

Additionally, this approach opens up opportunities for offering a range of customizations. For instance, it could allow for the dynamic alteration of button texts and icons based on the OIDC client in use, offering more tailored prompts such as Login with Github or Login with Okta rather than a generic Login with SSO. Furthermore, it could provide administrators with the flexibility to manage authentication methods more effectively, such as disabling username/password logins, enforcing multi-factor authentication (MFA), or even disabling the password reset functionality.

I'm would love to hear your perspective on this. If you have a strong preference for not keeping it generic and would rather we switch to a more specific endpoint like `/dex/enabled`, I'm open to making that adjustment.

Reply:
While it's purely semantic, REST guidelines advocate for the use of nouns rather than verbs in endpoint URLs, aligning with RESTful API conventions. Naming the endpoint `capabilities` adheres to this best practice :) 

<img width="830" alt="Screenshot 2024-03-27 at 12 23 18" src="https://github.com/litmuschaos/litmus/assets/10204970/83eb1ab1-d800-4ccb-a799-2b45209e4e5e">

Reply:
Thanks for the detailed explanation :) I completely agree 👍  I will review the remaining soon.